### PR TITLE
Move a few needed functions in `Correspondent`

### LINKED ISSUE #217: Add `Correspondent` interface

### PR DESCRIPTION
**Pull request by LunarX**
Closes #217

Depends on #205

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt [hunk 1]
```diff
@@ -38,13 +38,13 @@ object UiUtils {
...
         correspondent: Correspondent,
+        nameTextView: TextView,
+        emailTextView: TextView? = null,
     ) = with(correspondent) {
         when {
-            email.isMe() -> {
-                val context = nameTextView.context ?: emailTextView?.context!!
+            isMe() -> {
+                val context = nameTextView.context
                 nameTextView.text = context.getString(R.string.contactMe)
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt:48]: Why not using  `nameTextView.setText(R.string.contactMe)` ?

File: app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt [hunk 2]
```diff
@@ -38,13 +38,13 @@ object UiUtils {
     fun formatUnreadCount(unread: Int) = if (unread >= 100) "99+" else unread.toString()
 
     fun fillInUserNameAndEmail(
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt:40]: Maybe move this to Correspondant interface to use it as an extension function ?

File: app/src/main/java/com/infomaniak/mail/utils/Extensions.kt [hunk 3]
```diff
@@ -82,7 +84,13 @@ fun View.setMargins(left: Int = 0, top: Int = 0, right: Int = 0, bottom: Int = 0
 
 fun String.isEmail(): Boolean = Patterns.EMAIL_ADDRESS.matcher(this).matches()
 
-fun String.isMe(): Boolean = AccountUtils.currentUser?.email == this
+fun ImageView.loadAvatar(
+    correspondent: Correspondent,
+    imageLoader: ImageLoader = ImageLoader.Builder(context).build(),
+): Disposable = with(correspondent) {
+    val initials = (name.ifBlank { email }).firstOrEmpty().toString().uppercase()
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/Extensions.kt:91]: Use `getNameOrEmail()` instead.

File: app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt (hunk 2 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt:40]: 🙅

Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt:40]: Because this modifies the UI, this won't go in Correspondent

**Review by KevinBoulongne — APPROVED** [review]

**Review by FabianDevel — APPROVED** [review]

**Review by JorisBodin — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/utils/Extensions.kt [hunk 4]
```diff
@@ -82,7 +84,13 @@ fun View.setMargins(left: Int = 0, top: Int = 0, right: Int = 0, bottom: Int = 0
 
 fun String.isEmail(): Boolean = Patterns.EMAIL_ADDRESS.matcher(this).matches()
 
-fun String.isMe(): Boolean = AccountUtils.currentUser?.email == this
+fun ImageView.loadAvatar(
+    correspondent: Correspondent,
+    imageLoader: ImageLoader = ImageLoader.Builder(context).build(),
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/utils/Extensions.kt:89]: Useless

**Review by sirambd — APPROVED** [review]

**Pull request merged**